src/Symfony/Component/Messenger/Transport/Serialization/PhpSerializer.php

```diff
@@ -93,6 +93,10 @@ private function safelyUnserialize(string $contents): Envelope
         try {
             /** @var Envelope */
             $envelope = unserialize($contents);
+        } catch (\Throwable $e) {
+            $message = ($e instanceof \UnserializationFailedException ? $e->getPrevious() : $e)->getMessage();
+
+            throw new MessageDecodingFailedException($message, 0, $e);
```

Comment:
Here, we fix the test suite by unwrapping only the error message. On not-found-classes, PHP 8.3 would throw a MessageDecodingFailedException that wraps an UnserializationFailedException that itself wraps the original MessageDecodingFailedException. Pre-8.3, last exception is thrown directly. We could unwrap that stack instead, but that'd just mean endorsing the BC break brought by the RFC.